lib/ui/compositing.dart

```diff
@@ -57,10 +57,7 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   ///
   /// See [pop] for details about the operation stack.
   EngineLayer pushTransform(Float64List matrix4) {
-    if (matrix4 == null)
-      throw new ArgumentError('"matrix4" argument cannot be null');
-    if (matrix4.length != 16)
-      throw new ArgumentError('"matrix4" must have 16 entries.');
+    _throwIfMatrix4IsInvalid(matrix4);
```

Comment:
Rather than creating `_throwIfMatrix4IsInvalid()` which duplicates the checks in `_matrix4IsValid()`, I think we should either (a) just do an if check on `_matrix4IsValid()` here and throw `ArgumentError` saying the matrix is invalid, or (b) just call `assert(_matrix4IsValid(matrix4))` here.

B is more in keeping with our existing data validation in the framework, which is to catch these types of problems at development time and optimize out the checks in release builds.